tests/ui/let-else/let-else-binding-explicit-mut-borrow.stderr

```diff
@@ -2,7 +2,11 @@ error[E0596]: cannot borrow data in a `&` reference as mutable
   --> $DIR/let-else-binding-explicit-mut-borrow.rs:7:37
    |
 LL |     let Some(n): &mut Option<i32> = &mut &Some(5i32) else {
-   |                                     ^^^^^^^^^^^^^^^^ cannot borrow as mutable
+   |                                     ^^^^^-----------
+   |                                          |
+   |                                          cannot borrow as mutable
+   |
+   = help: `&Some(5i32)` is of type `&std::option::Option<i32>`, which is an immutable reference
```

Comment:
If the expression contains newlines, would this help message contain newlines too? Maybe you could just use the span?

Reply:
yeah, ideally we wouldn't use span_to_snippet and just use span_help and say "this expression"

Reply:
fixed, thanks!